### PR TITLE
 tracing: set worker jobs trace sample rate probability to 2 by default (configurable)

### PR DESCRIPTION
**Pull request by Strum355**
Dem some big numbers, lets make em smol :chart_with_downwards_trend: 

image

## Test plan

N/A, just changing a non critical sampling probability

### CODE REVIEW

**Review by bobheadxi — APPROVED** [review]
nice, thank you :)

File: enterprise/cmd/executor/internal/run/util.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ func makeWorkerMetrics(queueName string) workerutil.WorkerMetrics {
 		Registerer: prometheus.DefaultRegisterer,
 	}
 
-	return workerutil.NewMetrics(observationContext, "executor_processor",
+	return workerutil.NewMetrics(observationContext, "executor_processor", func(job workerutil.Record) bool { return true },
```
Comment (Strum355) [inline enterprise/cmd/executor/internal/run/util.go:190]: No executor tracing...yet :smugcat:

File: internal/workerutil/observability.go [hunk 2]
```diff
@@ -54,8 +59,8 @@ func WithDurationBuckets(buckets []float64) MetricOption {
 //   - {prefix}_error_total: number of handler operations resulting in an error
 //   - {prefix}_handlers: the number of active handler routines
 //
-// The given labels are emitted on each metric.
-func NewMetrics(observationContext *observation.Context, prefix string, opts ...MetricOption) WorkerMetrics {
+// The given labels are emitted on each metric. If traceSampler is unset, traces will have a 1 in 2 probability of being sampled.
+func NewMetrics(observationContext *observation.Context, prefix string, traceSampler func(job Record) bool, opts ...MetricOption) WorkerMetrics {
```
Comment (bobheadxi) [inline internal/workerutil/observability.go:63]: @Strum355 thought - could `traceSampler` be configured through `MetricOption`?

```go
	options := &metricOptions{
		durationBuckets: prometheus.DefBuckets,
		sampler: func(job Record) bool { return rand.Int31()%2 == 0 } 
	}
```

Then:

```go
workerutil.NewMetrics(ctx, prefix, workerutil.WithSampler(...)
```

Comment (bobheadxi) [reply internal/workerutil/observability.go:63]: It's a bit strange but `WorkerMetrics` as a whole already seems to handle tracing logic so this isn't too far off the mark

Comment (Strum355) [reply internal/workerutil/observability.go:63]: I was considering something like that, may go for it in the end 🤷

Comment (Strum355) [reply internal/workerutil/observability.go]: Updated

**Pull request merged**